build-scripts/build_rule_playbooks.py

```diff
@@ -0,0 +1,335 @@
+        snippet_str = snippet_str.replace("@ANSIBLE_ENSURE_PLATFORM@", "")
+        snippet_str = snippet_str.replace("@ANSIBLE_PLATFORM_CONDITION@",
+                                          "true")
+        snippet_str = snippet_str.replace("@RULE_ID@", "rule_id")
+        snippet_yaml = ssg.yaml.ordered_load(snippet_str)
+        # end of workaround
+
+        play_tasks, play_vars = self.get_data_from_snippet(
+            snippet_yaml, variables, refinements
+        )
+
+        if len(play_tasks) == 0:
+            raise ValueError(
+                "Ansible remediation for rule '%s' in '%s' "
+                "doesn't contain any task." %
+                (rule_id, snippet_path)
+            )
+
+        # play should be a dictionary, but to keep a logical order of elements
+        # we create the play as a list of tuples and later we convert it to
+        # an OrderedDict
+        play = [
+            ("name", rule_id),
+            ("hosts", "@@HOSTS@@"),
+            ("become", True),
+        ]
+        if len(play_vars) > 0:
+            play.append(("vars", play_vars))
+        if len(tags) > 0:
+            play.append(("tags", tags))
+        play.append(("tasks", play_tasks))
+        play_ordered_dict = OrderedDict(play)
+
+        playbook = [play_ordered_dict]
+        playbook_path = os.path.join(output_dir, rule_id + ".yml")
+        with open(playbook_path, "w") as playbook_file:
+            ssg.yaml.ordered_dump(
+                playbook, playbook_file, default_flow_style=False
+            )
+
+    def open_profile(self, profile_path):
+        """
+        Opens and parses profile at the given profile_path.
+        """
+        if not os.path.isfile(profile_path):
+            raise RuntimeError("'%s' is not a file!\n" % profile_path)
+        profile_id, ext = os.path.splitext(os.path.basename(profile_path))
+        if ext != ".profile":
+            raise RuntimeError(
+                "Encountered file '%s' while looking for profiles, "
+                "extension '%s' is unknown. Skipping..\n"
+                % (profile_path, ext)
+            )
+
+        profile = ssg.build_yaml.Profile.from_yaml(profile_path, self.env_yaml)
+        if not profile:
+            raise RuntimeError("Could not parse profile %s.\n" % profile_path)
+        return profile
+
+    def create_playbooks_for_all_rules(self, profile, variables):
+        """
+        Creates a Playbook for each rule selected in a profile from tasks
+        extracted from snippets. Created Playbooks are parametrized by
+        variables according to profile selection. Playbooks are written into
+        a new subdirectory in output_dir.
+        """
+        profile_rules, profile_refines = self.get_profile_selections(profile)
+
+        profile_playbooks_dir = os.path.join(self.output_dir, profile.id_)
+        os.makedirs(profile_playbooks_dir)
+
+        for snippet in os.listdir(self.input_dir):
```

Comment:
It should be faster to cycle through `profile_rules` instead of all snippets found in directory.

Reply:
Fixed